ViV-Wallet-api/src/main/java/com/invivoo/vivwallet/api/domain/action/ActionService.java

```diff
@@ -22,20 +27,17 @@ public ActionService(ActionRepository actionRepository, LynxConnector lynxConnec
         this.lynxConnector = lynxConnector;
     }
 
-    public List<Action> findAllById(Collection<Long> actionIds) {
-        return actionRepository.findAllById(actionIds);
-    }
-
     public List<Action> findAllByOrderByDateDesc() {
         return actionRepository.findAllByIsDeletedFalseOrderByDateDesc();
     }
 
     public List<Action> findAllByAchieverOrderByDateDesc(User achiever) {
-        return actionRepository.findAllByAchieverOrderByDateDesc(achiever);
-    }
-
-    public List<Action> findAllByAchieverAndPaymentIsNullOrderByDateAsc(User achiever) {
-        return actionRepository.findAllByAchieverAndPaymentIsNullOrderByDateAsc(achiever);
+        LocalDateTime startValueDate = Optional.ofNullable(achiever.getVivInitialBalanceDate()).orElse(
+                LocalDateTime.MIN);
+        ArrayList<Action> actions = new ArrayList<>(actionRepository.findAllByAchieverAndValueDateAfter(achiever, startValueDate));
```

Comment:
List<Action> actions = actionRepository.findAllByAchieverAndValueDateAfter(achiever, startValueDate);

Reply:
O_o thought jpa repo was returning unmodifiableList ^^I'll do it

Reply:
Not doing it though, I prefer considering return of a method unmodifiable, modifying it can have some weird side effects